pkg/provider/kubernetes/gateway/kubernetes.go

```diff
@@ -0,0 +1,867 @@
+						router.Service = serviceName
+					}
+
+					if router.Service != "" {
+						routerKey = provider.Normalize(routerKey)
+						routers[routerKey] = &router
+					}
+				}
+			}
+
+			if len(listenerStatuses[i].Conditions) == 0 {
+				// GatewayConditionReady "Ready", GatewayConditionReason "ListenerReady"
+				listenerStatuses[i].Conditions = append(listenerStatuses[i].Conditions, metav1.Condition{
+					Type:               string(v1alpha1.ListenerConditionReady),
+					Status:             metav1.ConditionTrue,
+					LastTransitionTime: metav1.Now(),
+					Reason:             "ListenerReady",
+					Message:            "No error found",
+				})
+			} else {
+				listenerInError = true
+			}
+		}
+
+		gatewayStatus.Listeners = listenerStatuses
+
+		if listenerInError {
+			// GatewayConditionReady "Ready", GatewayConditionReason "ListenersNotValid"
+			gatewayStatus.Conditions = append(gatewayStatus.Conditions, metav1.Condition{
+				Type:               string(v1alpha1.GatewayConditionReady),
+				Status:             metav1.ConditionFalse,
+				LastTransitionTime: metav1.Now(),
+				Reason:             string(v1alpha1.GatewayReasonListenersNotValid),
+				Message:            "All Listeners must be valid",
+			})
+
+			err := client.UpdateGatewayStatus(gateway, gatewayStatus)
+			if err != nil {
+				logger.Debugf("An error occurred while updating gateway status: %v", err)
+			}
+
+			continue
+		}
+
+		// update "Scheduled" status with "ResourcesAvailable" reason
+		gatewayStatus.Conditions = append(gatewayStatus.Conditions, metav1.Condition{
+			Type:               string(v1alpha1.GatewayConditionScheduled),
+			Status:             metav1.ConditionTrue,
+			LastTransitionTime: metav1.Now(),
+			Reason:             "ResourcesAvailable",
+			Message:            "Found matching HTTPRoutes",
+		})
+
+		// update "Ready" status with "ListenersValid" reason
+		gatewayStatus.Conditions = append(gatewayStatus.Conditions, metav1.Condition{
+			Type:               string(v1alpha1.GatewayConditionReady),
+			Status:             metav1.ConditionTrue,
+			LastTransitionTime: metav1.Now(),
+			Reason:             "ListenersValid",
+			Message:            "All listeners are valid",
+		})
+
+		err := client.UpdateGatewayStatus(gateway, gatewayStatus)
+		if err != nil {
+			logger.Debugf("An error occurred while updating gateway status: %v", err)
+		}
+
+		if len(routers) > 0 {
+			for k, router := range routers {
+				// TODO manage duplicate key
+				conf.HTTP.Routers[k] = router
+			}
+		}
+
+		if len(services) > 0 {
+			for k, service := range services {
+				// TODO manage duplicate key
+				conf.HTTP.Services[k] = service
+			}
+		}
+
+		if len(tlsConfigs) > 0 {
+			if conf.TLS == nil {
+				conf.TLS = &dynamic.TLSConfiguration{}
+			}
+			conf.TLS.Certificates = append(conf.TLS.Certificates, getTLSConfig(tlsConfigs)...)
+		}
+	}
+
+	return conf
+}
+
+func hostRule(httpRouteSpec v1alpha1.HTTPRouteSpec) string {
+	if len(httpRouteSpec.Hostnames) > 0 {
```

Comment:
I think this check is not necessary because the `hostRule` will be empty as we will not iterate on the `httpRouteSpec.Hostnames`.